babel/index.js

```diff
@@ -89,6 +90,96 @@ BabelClient.prototype.headTargetFeed = function headTargetFeed(target, token, pa
     });
 };
 
+BabelClient.prototype.getEntireTargetFeed = async function (target, token, hydrate, callback) {
+    let callbackError = undefined;
+    if(!target){
+        callbackError = Error('Missing target');
+        callbackError.http_code = 400;
+        return callback(callbackError);
+    }
+    if(!token){
+        callbackError = Error('Missing Persona token');
+        callbackError.http_code = 400;
+        return callback(callbackError);
+    }
+
+    let results = {
+        annotations: []
+    };
+    const perPage = 1000;
+    let currentPage = 0;
+    let isFinalPage = false;
+
+    do {
+        this.debug(`getTargetFeedNoLimit fetching page' - ${currentPage}`);
+
+        const queryString = this._queryStringParams({
+            limit: perPage,
+            offset: currentPage * perPage,
+        })
+
+        const requestOptions = {
+            json: true,
+            simple: false,
+            resolveWithFullResponse: true,
+            method: 'GET',
+            url:  this._getBaseURL()
+                + '/feeds/targets/'
+                + md5(target)
+                + '/activity/annotations'
+                + ((hydrate === true) ? '/hydrate' : '')
+                + (!_.isEmpty(queryString) ? '?'+queryString : ''),
+            headers: {
+                'Accept': 'application/json',
+                'Authorization':'Bearer '+ token,
+                'Host': this.config.babel_hostname,
+            },
+        };
+
+        this.debug(JSON.stringify(requestOptions));
+
+        try {
+            const { 
+                body: {
+                    feed_length,
+                    annotations,
+                    userProfiles,
+                    error,
+                    error_description,
+                }, 
+                ...response
+            } = await rpn(requestOptions);
+            
+            if (error) {
+                callbackError = new Error(error_description);
+                callbackError.http_code = response.statusCode || 404;
+            } else {
+                results = {
+                    ...results,
+                    feed_length,
+                    annotations: [
+                        ...results.annotations,
+                        ...annotations
+                    ],
+                    userProfiles: {
+                        ...results.userProfiles,
+                        ...userProfiles
+                    }
+                }
+                isFinalPage = annotations.length < perPage;
+            }
+
+        } catch (error) {
+            this.error(`Error fetching babel feed ${JSON.stringify(error)}`);
+            callbackError = error;
+        }
+
+        currentPage = currentPage + 1;
+    } while (!isFinalPage && callbackError === undefined);
```

Comment:
Can't remember the last time I saw a `do { } while ();` loop! Any reason why we can't use a `while () { }` loop? Much easier to reason about when the condition is at the start of the block

Reply:
I changed it based on feedback from @pogotc, he was nervous about using a while loop.

Reply:
Yeah I've just read his comments, and I think the previous concerns were more with the condition in the loop. I don't think our linter prefers one type over the other. So my preference of `while` > `do while` is not worth changing here.

Reply:
I don't mind either way, I suppose with a `do while` loop 9/10 the condition starts as false a stays that way so you never ever the loop? With a while loop we have to enter the loop every time?

As for linting we don't think we lint this project do we, I tried running it and it failed on something that was changed like a year ago?

Reply:
I think a `do while` fits this use case reasonably well as it's always guaranteed to run the code inside at least once which is what we want. That said, with the condition as it currently is I'm not opposed to it moving to the top and becoming a regular while loop. I was mostly uncomfortable with having a page count inside the while loop and then breaking out of it as it felt a bit more prone to hitting an infinite loop.